packages/grafana-data/src/transformations/transformers/groupingToMatrix.test.ts

```diff
@@ -0,0 +1,170 @@
+import {
+  ArrayVector,
+  DataTransformerConfig,
+  DataTransformerID,
+  Field,
+  FieldType,
+  toDataFrame,
+  transformDataFrame,
+} from '@grafana/data';
+import { GroupingToMatrixTransformerOptions, groupingToMatrixTransformer } from './groupingToMatrix';
+import { mockTransformationsRegistry } from '../../utils/tests/mockTransformationsRegistry';
+
+describe('Grouping to Matrix', () => {
+  beforeAll(() => {
+    mockTransformationsRegistry([groupingToMatrixTransformer]);
+  });
+
+  it('generates Matrix with default fields', async () => {
+    const cfg: DataTransformerConfig<GroupingToMatrixTransformerOptions> = {
+      id: DataTransformerID.groupingToMatrix,
+      options: {},
+    };
+
+    const seriesA = toDataFrame({
+      name: 'A',
+      fields: [
+        { name: 'Time', type: FieldType.time, values: [1000, 1001, 1002] },
+        { name: 'Value', type: FieldType.number, values: [1, 2, 3] },
+      ],
+    });
+
+    await expect(transformDataFrame([cfg], [seriesA])).toEmitValuesWith((received) => {
+      const processed = received[0];
+      const expected: Field[] = [
+        {
+          name: 'Time\\Time',
+          type: FieldType.string,
```

Comment:
Are there any reason why we don't want to keep the `FieldType.time` here?

Reply:
Not really, but since it's used for "Column/Rows" names I thought forcing String made sense

Reply:
Ok ok, sounds good!